noodles-fasta/src/fai/indexer.rs

```diff
@@ -0,0 +1,240 @@
+//! Creates a index for a FASTA file
+
+use std::convert::TryInto;
+use std::error::Error;
+use std::fmt;
+use std::io;
+
+use super::Record;
+use crate::reader::{DEFINITION_PREFIX, NEWLINE};
+use crate::record::definition::{Definition, ParseError};
+
+use memchr::memchr;
+
+/// A index builder
+pub struct Indexer<R> {
+    inner: R,
+    offset: u64,
+}
+
+impl<R> Indexer<R>
+where
+    R: io::BufRead,
```

Comment:
```suggestion
    R: BufRead,
```

Import `std::io::BufRead`.